internal/sdkio/ringbuffer.go

```diff
@@ -0,0 +1,100 @@
+package sdkio
+
+import (
+	"io"
+)
+
+// RingBuffer struct satisfies io.ReadWrite
+// interface.
+//
+// ReadBuffer is a revolving buffer data structure
+// which can be used to store snapshots of data in a
+// revolving window.
+type RingBuffer struct {
+	slice []byte
+	start int
+	end   int
+	size  int
+}
+
+// New method takes in the capacity as an int
+// and returns a RingBuffer.
+func New(capacity int) *RingBuffer {
+	slice := make([]byte, capacity, capacity)
+	ringBuf := RingBuffer{
+		slice: slice,
+		start: 0,
+		end:   0,
+		size:  0,
+	}
+	return &ringBuf
+}
+
+// Write method inserts the elements
+// in the byte slice, and return
+// the number of bytes written and an error
+//
+// This satisfies io.Writer interface.
```

Comment:
Nit don't have to wrap comment so signfiicantly at least 80 characters is fine :)